lib/data/alexa_similar_sites.ex

```diff
@@ -31,19 +36,46 @@ defmodule Data.AlexaSiteInfo do
 
   defp update_caches(url) do
     Logger.info("Cache miss - '#{url}' - Alexa info - Fetching.")
-    {alexa_ranks, similar_sites_with_similarity_ranks} = fetch_data_from_alexa(url)
-    Util.PersistentCache.put_many(@ranks, alexa_ranks)
-    Util.PersistentCache.put(@similar, url, similar_sites_with_similarity_ranks)
+
+    case fetch_data_from_alexa(url) do
+      :error ->
+        Logger.info("Failed to find site.")
+        :error
+
+      {:ok, {alexa_ranks, similar_sites_with_similarity_ranks}} ->
+        Util.PersistentCache.put_many(@ranks, alexa_ranks)
+        Util.PersistentCache.put(@similar, url, similar_sites_with_similarity_ranks)
+        :ok
+    end
   end
 
-  defp fetch_data_from_alexa(host) do
-    [main_site_info | similar_sites_info] =
-      HTTPoison.get!(@site_info_url <> host, %{}, recv_timeout: @timeout_ms)
-      |> Map.get(:body)
+  defp get_alexa_data(%HTTPoison.Response{body: "{\"errors\":[\"Please enter a valid site.\"]}\n"}) do
```

Comment:
might be cleaner to match on the HTTP status code here (probably a `400` [maybe `500`?]) rather than hard matching the response body

Reply:
It's a 200

Reply:
dang, okay